### ISSUE TITLE
D2X-Rebirth: Selecteable static in right cockpit view

### ISSUE THREAD

**Issue by zicodxx** [dxx-rebirth/dxx-rebirth#199]
If you decide to cycle through all the available cockpit window views using SHIFT+F2, before switching back to the Weapon display, the game will shortly display static as if a missile is just about to explode. 

This only applies to the right window (secondary weapons). If cycling the primary window (SHIFT+F1) this does not happen.

**Comment by Mako88** [comment]
Is this in some way related to the first screenshot in #6 ?

**Comment by zicodxx** [comment]
No, it's not related to that. 

So assume you are in cockpit and see your weapon info, if you press SHIFT+F2, the following should be visible in that order:
Rear View -> Guide-Bot -> Weapon Info
But what it does show is:
Rear View -> Guide-Bot -> Static followed by Weapon Info

I have bisected this up to 32b9202e1f5443a32b59ac3a8808f5555d14600f but now I ran out of time so I have to continue digging deeper on Friday.

**Comment by zicodxx** [comment]
I committed  5d2b2891ad39c6d29b391b4bf9b5598b6df5b71e to fix this issue. This code adds an extra condition to trigger WBU_STATIC which also wasn't present in the original either but it prevents per-frame calls of do_cockpit_window_view which I think is good and it does the job fine. 
To be frank I would have preferred to trace where exactly 32b9202e1f5443a32b59ac3a8808f5555d14600f caused the accidental triggers of WBU_STATIC... so I may revisit this one again if I have a bit more time.

**Comment by Mako88** [comment]
Since you'd like to revisit it, I won't close this, but I'm going to remove it from the 0.60 milestone so we can more precisely track exactly what still needs to be done as feedback from the RC comes in.

**Comment by kreatordxx** [comment]
Aha - I managed to track this down. The reason 32b9202 causes the accidental trigger of `WBU_STATIC` is because of a removal of `{}`'s, so that the code block calling `do_cockpit_window_view(1,WBU_STATIC)` is called every time `choose_missile_viewer()` is false. Before, it would only call that if 

```
PlayerCfg.MissileViewEnabled && Missile_viewer->type!=OBJ_NONE && Missile_viewer->signature == Missile_viewer_sig
```

is false, and if there **is** a missile viewer. This above check was made redundant by 32b9202.

Therefore, @zicodxx's fix 5d2b289 appears to be correct.

**Comment by vLKp** [comment]
Kreator identified a reasonable explanation for the problem, so it appears there is nothing further to do here.

Closing.
